### PR TITLE
fix: chrome should not wrap mid-line

### PR DESCRIPTION
**Pull request by nperez0111**
Partially reverts commit "e011cf4863e06b6eb07985f379f7750d76b091ee"

This addresses a bug introduced by https://github.com/TypeCellOS/BlockNote/pull/1422 which caused a regression where chrome in some instances would break text wrapping because of the pseudo element's presence in the document.

The pseduo element is now removed, moving back to the original border color approach with an additional fix specifically applied to firefox that will cause the element not to wrap when whitespace is following text

### CODE REVIEW

**Comment by nperez0111** [comment]
I went back on this approach again. I found that the pseudo element can use a border so that it does not take space within the content, but still have a width to allow mouse events to work on the element. Seems to be working well in firefox & chrome in the various scenarios we have identified.

Will want to look into this in the morning though....

File: packages/core/src/extensions/Collaboration/createCollaborationExtensions.ts [hunk 1]
```diff
@@ -77,9 +77,7 @@ export const createCollaborationExtensions = (collaboration: {
       labelElement.setAttribute("style", `background-color: ${user.color}`);
       labelElement.insertBefore(document.createTextNode(user.name), null);
 
-      cursorElement.insertBefore(document.createTextNode("\u2060"), null); // Non-breaking space
```
Comment (nperez0111) [inline packages/core/src/extensions/Collaboration/createCollaborationExtensions.ts:80]: We now let CSS insert these &nbsp;

**Comment by nperez0111** [comment]
https://github.com/user-attachments/assets/0e7a5199-f0a0-433c-84aa-eec34d04e847

Left Firefox, Middle Chrome, Right Safari

I think I've got all of the edge cases figured out for the major browsers here. They each needed their own hacks. I went with the most chrome native implementation (using the pseudo element with a border) and then added the hacks for firefox & safari behavior on top with comments on why.

Hopefully this will resolve it

File: packages/core/src/editor/editor.css [hunk 2]
```diff
@@ -79,21 +79,32 @@ Tippy popups that are appended to document.body directly
...
+@-moz-document url-prefix() {
+  .collaboration-cursor__caret::before {
+    display: inline-block;
+  }
+}
+
+/* Add nbsp; to each side of the caret */
+.collaboration-cursor__caret::after,
+.collaboration-cursor__caret::before {
+  content: "⁠";
```
[the remark this answers is not in the document]
Comment (nperez0111) [reply packages/core/src/editor/editor.css:107]: word joiner is there. I'll update the comment with the unicode to make it clear

**Comment by nperez0111** [comment]
> There's a little bit of whitespace now again right?

There is but the negative margin slurps it so it becomes effectively 0px wide (though Safari seems to not agree with that unless position: relative is set).

> shall we comment the scenarios as to why this was complicated? (I hope we're not touching this code for a while, but maybe good to explain that changes need to be tested on all browsers and both the long lines + whitespace end of line scenarios?)

Yea, I'll add a here be dragons comment

**Review by YousefED — APPROVED** [review]
